File: backend/databasehandler/src/main/java/com/pixtranslator/backend/databasehandler/repository/GermanWordRepository.java
package com.pixtranslator.backend.databasehandler.repository;

import com.pixtranslator.backend.databasehandler.model.GermanWord;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface GermanWordRepository extends CrudRepository<GermanWord, String> {
  Optional<GermanWord> findByWordIgnoreCase(String word);
}

File: backend/databasehandler/src/main/kotlin/com/pixtranslator/backend/databasehandler/repository/GermanWordRepository.kt
package com.pixtranslator.backend.databasehandler.repository

import com.pixtranslator.backend.databasehandler.model.GermanWord
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface GermanWordRepository : CrudRepository<GermanWord, String> {
  fun findByWordIgnoreCase(word: String?): Optional<GermanWord>
}